d3a_api_client/cli.py

```diff
@@ -91,21 +91,22 @@ def run(base_setup_path, setup_module_name, username, password, domain_name, web
         os.environ["API_CLIENT_PASSWORD"] = password
 
     os.environ["API_CLIENT_RUN_ON_REDIS"] = "true" if run_on_redis else "false"
-    if simulation_config_path is not None and not run_on_redis:
-        config_file_path = create_simulation_config_path(base_setup_path, simulation_config_path)
-        config = read_simulation_config_file(config_file_path)
-        os.environ["API_CLIENT_DOMAIN_NAME"] = config["domain_name"]
-        os.environ["API_CLIENT_WEBSOCKET_DOMAIN_NAME"] = config["web_socket_domain_name"]
-        os.environ["API_CLIENT_SIMULATION_ID"] = config["uuid"]
-    else:
-        os.environ["API_CLIENT_DOMAIN_NAME"] = domain_name \
-            if domain_name else domain_name_from_env()
-        os.environ["API_CLIENT_WEBSOCKET_DOMAIN_NAME"] = web_socket \
-            if web_socket else websocket_domain_name_from_env()
-        os.environ["API_CLIENT_SIMULATION_ID"] = simulation_id \
-            if simulation_id else simulation_id_from_env()
-
-    validate_settings_are_set_before_launch()
+    if not run_on_redis:
+        if simulation_config_path is not None:
+            config_file_path = create_simulation_config_path(base_setup_path, simulation_config_path)
+            config = read_simulation_config_file(config_file_path)
+            os.environ["API_CLIENT_DOMAIN_NAME"] = config["domain_name"]
+            os.environ["API_CLIENT_WEBSOCKET_DOMAIN_NAME"] = config["web_socket_domain_name"]
+            os.environ["API_CLIENT_SIMULATION_ID"] = config["uuid"]
+        else:
+            os.environ["API_CLIENT_DOMAIN_NAME"] = domain_name \
+                if domain_name else domain_name_from_env()
+            os.environ["API_CLIENT_WEBSOCKET_DOMAIN_NAME"] = web_socket \
+                if web_socket else websocket_domain_name_from_env()
+            os.environ["API_CLIENT_SIMULATION_ID"] = simulation_id \
+                if simulation_id else simulation_id_from_env()
+
+        validate_settings_are_set_before_launch()
```

Comment:
This should be unindented

Reply:
@Hassan754  not really, have a loot into the function. All the settings are not needed for the redis case.

Reply:
I should remove `API_CLIENT_RUN_ON_REDIS` from `settings_list` though

Reply:
https://github.com/gridsingularity/d3a-api-client/pull/129/commits/793dac45dd85889b2692d0eaa77c0688d8b9e112